types/jsreport-core/index.d.ts

```diff
@@ -7,90 +7,107 @@
 /// <reference types="node" />
 
 declare namespace JsReport {
-	type helpers = string | { [fun: string]: (...args: any[]) => any };
-
-	interface RenderOptions {
-		template: {
-			content: string;
-			engine: 'jsrender' | 'handlebars' | 'ejs' | 'jade' | string;
-			helpers?: helpers;
-			recipe: 'phantom-pdf' | 'electron-pdf' | 'text' | 'xlsx' | 'html-to-xlsx' | 'phantom-image' | 'html-to-text' | 'fop-pdf' | 'client-html' | 'wrapped-html' | 'wkhtmltopdf' | string;
-		};
-		data?: any;
-	}
-
-	interface Report {
-		content: Buffer;
-		stream: NodeJS.ReadableStream;
-		headers: {
-			[header: string]: string | number | boolean;
-		};
-	}
-
-	interface Request {
-		template: {
-			content: string;
-		};
-	}
-
-	// interface Response {
-	// 	// todo
-	// }
-
-	type Response = any;
-
-	interface Listener {
-		add(type: string, callback: (req: Request, res: Response, err: any) => void): void;
-	}
-
-	interface Logger {
-		add(logger: any, options?: {
-			level: 'debug' | 'info' | 'log' | 'warn' | 'error';
-		}): void;
-	}
-
-	interface Collection {
-		find(query: {
-			[field: string]: any;
-		}): Promise<any>;
-	}
-
-	interface DocumentStore {
-		collection(options: string): Collection;
-	}
-
-	interface JsReporter {
-		afterRenderListeners: JsReport.Listener;
-		afterTemplatingEnginesExecutedListeners: JsReport.Listener;
-		beforeRenderListeners: JsReport.Listener;
-		documentStore: JsReport.DocumentStore;
-		initializeListeners: JsReport.Listener;
-		logger: JsReport.Logger;
-		validateRenderListeners: JsReport.Listener;
-		init(): Promise<void>;
-		render(options: JsReport.RenderOptions): Promise<JsReport.Report>;
-		use(extension: any): any;
-	}
+  type Helpers = string | { [fun: string]: (...args: any[]) => any };
+
+  const enum Engine {
```

Comment:
We do. Try to remove it and use 

```js
import * as JsReport from "jsreport-core";
JsReport.Engine.None
```

You get error that `JsReport.Engine` is undefined.

Reply:
https://github.com/taoqf/jsreport-core/commit/22cb047de7997c5d534f1333d9829c4e266479e3
but I have not got that error

Reply:
You should get that error. The const instruct typescript to transpile `JsReport.Engine.None` into string constant. Without const it transpiles into `JsReport.Engine.None` and expects this is provided by the underlying library. But it is not, there are no such enums inside jsreport core. It expects string constant on the input.

I'm not able to build the tsc workspace you have here, unfortunately I cannot verify it.
Does the const otherwise cause any harm? If not, I would send PR to definitely typed so we can close this.